network.go

```diff
@@ -1156,33 +1156,34 @@ func (n *network) createEndpoint(name string, options ...EndpointOption) (Endpoi
 			ep.releaseAddress()
 		}
 	}()
-
-	if err = n.addEndpoint(ep); err != nil {
+	//Moving updateToSTore before calling addEndpoint so that we shall clean up VETH interfaces in case
```

Comment:
Very minor stylistic nit: I think typical convention for comments is to add a space between `//` and the comment line. So something like `// Moving updateToStore` would be ideal. Same applies to the line below.